### PR TITLE
[QI2-1119] Support running in an existing event loop (i.e. jupyter notebook)

### PR DESCRIPTION
**Pull request by napocornejo**

### CODE REVIEW

**Comment by eendebakpt** [comment]
@napocornejo Tested and works under spyder!

File: qiskit_quantuminspire/utils.py [hunk 1]
```diff
@@ -10,3 +14,17 @@ def is_coupling_map_complete(coupling_map: CouplingMap) -> bool:
...
+        loop = asyncio.get_running_loop()
+    except RuntimeError:  # 'RuntimeError: There is no current event loop...'
+        loop = None
+
+    if loop and loop.is_running():
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            future = executor.submit(asyncio.run, async_function)
+            return future.result()
+    else:
+        return asyncio.run(async_function)
```
Comment (eliasKA) [inline qiskit_quantuminspire/utils.py:30]:
```suggestion
    try:
        loop = asyncio.get_running_loop()
        with concurrent.futures.ThreadPoolExecutor() as executor:
            return executor.submit(asyncio.run, async_function).result()
    except RuntimeError:
        return asyncio.run(async_function)
```

**Review by eliasKA — APPROVED** [review]
Looks good ✅, one small suggestion for possibly rewriting `run_async` method but feel free to continue without it.

File: qiskit_quantuminspire/utils.py (hunk 1 above)
Comment (napocornejo) [reply qiskit_quantuminspire/utils.py]: good suggestion. Taken.

**Review by eliasKA — DISMISSED** [review]

**Review by eliasKA — APPROVED** [review]

**Pull request merged**